### ISSUE TITLE
nvda randomly crashes when trying to open NVDA menu

### ISSUE THREAD

**Issue by zstanecic** [nvaccess/nvda#8086]
steps to reproduce:
1. try to open the NVDA menu.
2. you will hear windows error sound.
3. nvda will restart itself

this can happen three times. 
it happens from the wx next snaps.
 cannot catch the logs, installed version

**Comment by leonardder** [comment]
@zstanecic: Thanks for reporting.

Note that you could always find the most recent log and one log before in your temp folder, %temp% or c:\users\youruser\appdata\local\temp. When NVDA crashed and you run it again, the log file of the running instance is nvda.log. The log file of the crashed instance should be nvda-old.log at that time.

**Comment by Andre9642** [comment]
Hello,
I am sorry; but I think BrailleExtender is the cause of the problem... Have you tried with all addons disabled?
Indeed, when I open the NVDA menu, I have also this error from time to time, I don't know why for now... :/

```
DEBUGWARNING - Python warning (06:02:28.755):
C:\Program Files (x86)\NVDA\library.zip\gui\__init__.py:211: wxPyDeprecationWarning: Call to deprecated item. Use Remove instead.
CRITICAL - watchdog._crashHandler (06:02:29.073):
NVDA crashed! Minidump written to C:\Users\andre\AppData\Local\Temp\nvda_crash.dmp
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6816
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "braille.pyc", line 1965, in func
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6148
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "winInputHook.pyc", line 70, in hookThreadFunc
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 828
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "_UIAHandler.pyc", line 224, in MTAThreadFunc
  File "threading.pyc", line 614, in wait
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 13036
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
,  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "watchdog.pyc", line 82, in _watcher
  File "winKernel.pyc", line 208, in waitForSingleObject
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9244
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 81, in run
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9584
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 216, in run
  File "Queue.pyc", line 168, in get
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 2492
Stack trace:
  File "nvda.pyw", line 212, in <module>
  File "core.pyc", line 472, in main
  File "wx\core.pyc", line 2096, in MainLoop
  File "wx\core.pyc", line 3190, in <lambda>
  File "gui\__init__.pyc", line 157, in showGui
  File "gui\__init__.pyc", line 552, in onActivate
  File "watchdog.pyc", line 180, in _crashHandler
INFO - watchdog._crashHandler (06:02:31.273):
Python stack for thread 12924
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 221, in run
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 970, in autoreload_profile
INFO - watchdog._crashHandler (06:02:31.273):
Restarting due to crash
```

**Comment by zstanecic** [comment]
Hi,
With the addons disabled i cannot reproduce this.

Wysłane z aplikacji Poczta dla Windows 10

Od: André-Abush Clause
Wysłano: czwartek, 15 marca 2018 06:33
Do: nvaccess/nvda
DW: zstanecic; Mention
Temat: Re: [nvaccess/nvda] nvda randomly crashes when trying to open NVDAmenu (#8086)

Hello,
I am sorry; but I think BrailleExtender is the cause of the problem... Have you tried with all addons disabled?
Indeed, when I open the NVDA menu, I have also this error from time to time, I don't know why for now... :/
DEBUGWARNING - Python warning (06:02:28.755):
C:\Program Files (x86)\NVDA\library.zip\gui\__init__.py:211: wxPyDeprecationWarning: Call to deprecated item. Use Remove instead.
CRITICAL - watchdog._crashHandler (06:02:29.073):
NVDA crashed! Minidump written to C:\Users\andre\AppData\Local\Temp\nvda_crash.dmp
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6816
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "braille.pyc", line 1965, in func
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6148
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "winInputHook.pyc", line 70, in hookThreadFunc
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 828
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "_UIAHandler.pyc", line 224, in MTAThreadFunc
  File "threading.pyc", line 614, in wait
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 13036
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
,  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "watchdog.pyc", line 82, in _watcher
  File "winKernel.pyc", line 208, in waitForSingleObject
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9244
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 81, in run
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9584
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 216, in run
  File "Queue.pyc", line 168, in get
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 2492
Stack trace:
  File "nvda.pyw", line 212, in <module>
  File "core.pyc", line 472, in main
  File "wx\core.pyc", line 2096, in MainLoop
  File "wx\core.pyc", line 3190, in <lambda>
  File "gui\__init__.pyc", line 157, in showGui
  File "gui\__init__.pyc", line 552, in onActivate
  File "watchdog.pyc", line 180, in _crashHandler
INFO - watchdog._crashHandler (06:02:31.273):
Python stack for thread 12924
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 221, in run
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 970, in autoreload_profile
INFO - watchdog._crashHandler (06:02:31.273):
Restarting due to crash

**Comment by zstanecic** [comment]
No Leonard,
In my case old log disappears for some reason

Wysłane z aplikacji Poczta dla Windows 10

Od: Leonard de Ruijter
Wysłano: czwartek, 15 marca 2018 05:55
Do: nvaccess/nvda
DW: zstanecic; Mention
Temat: Re: [nvaccess/nvda] nvda randomly crashes when trying to open NVDAmenu (#8086)

@zstanecic: Thanks for reporting.
Note that you could always find the most recent log and one log before in your temp folder, %temp% or c:\users\youruser\appdata\local\temp. When NVDA crashed and you run it again, the log file of the running instance is nvda.log. The log file of the crashed instance should be nvda-old.log at that time.

**Comment by zstanecic** [comment]
Yes, confirmed,
Braille extender is the cause.
Have turned it off and it works. 

Wysłane z aplikacji Poczta dla Windows 10

Od: André-Abush Clause
Wysłano: czwartek, 15 marca 2018 06:33
Do: nvaccess/nvda
DW: zstanecic; Mention
Temat: Re: [nvaccess/nvda] nvda randomly crashes when trying to open NVDAmenu (#8086)

Hello,
I am sorry; but I think BrailleExtender is the cause of the problem... Have you tried with all addons disabled?
Indeed, when I open the NVDA menu, I have also this error from time to time, I don't know why for now... :/
DEBUGWARNING - Python warning (06:02:28.755):
C:\Program Files (x86)\NVDA\library.zip\gui\__init__.py:211: wxPyDeprecationWarning: Call to deprecated item. Use Remove instead.
CRITICAL - watchdog._crashHandler (06:02:29.073):
NVDA crashed! Minidump written to C:\Users\andre\AppData\Local\Temp\nvda_crash.dmp
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6816
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "braille.pyc", line 1965, in func
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 6148
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "winInputHook.pyc", line 70, in hookThreadFunc
INFO - watchdog._crashHandler (06:02:31.270):
Python stack for thread 828
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "_UIAHandler.pyc", line 224, in MTAThreadFunc
  File "threading.pyc", line 614, in wait
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 13036
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
,  File "threading.pyc", line 801, in __bootstrap_inner
  File "threading.pyc", line 754, in run
  File "watchdog.pyc", line 82, in _watcher
  File "winKernel.pyc", line 208, in waitForSingleObject
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9244
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 81, in run
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 9584
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\_eloquence.py", line 216, in run
  File "Queue.pyc", line 168, in get
  File "threading.pyc", line 340, in wait
INFO - watchdog._crashHandler (06:02:31.272):
Python stack for thread 2492
Stack trace:
  File "nvda.pyw", line 212, in <module>
  File "core.pyc", line 472, in main
  File "wx\core.pyc", line 2096, in MainLoop
  File "wx\core.pyc", line 3190, in <lambda>
  File "gui\__init__.pyc", line 157, in showGui
  File "gui\__init__.pyc", line 552, in onActivate
  File "watchdog.pyc", line 180, in _crashHandler
INFO - watchdog._crashHandler (06:02:31.273):
Python stack for thread 12924
Stack trace:
  File "threading.pyc", line 774, in __bootstrap
  File "threading.pyc", line 801, in __bootstrap_inner
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 221, in run
  File "C:\Users\andre\AppData\Roaming\nvda\addons\BrailleExtender\globalPlugins\brailleExtender\__init__.py", line 970, in autoreload_profile
INFO - watchdog._crashHandler (06:02:31.273):
Restarting due to crash

**Comment by ehollig** [comment]
As it seems Braille extender is the cause of this issue and not NVDA, I am going to close this issue.

**Comment by zstanecic** [comment]
the ussue is fixed in braille extender itself. the problem was, that it used a deprecated method
